sql/core/benchmarks/DataSourceReadBenchmark-jdk11-results.txt

```diff
@@ -2,322 +2,322 @@
+ParquetReader Vectorized: DataPageV2               1707           1713           9          6.1         162.8       4.9X
+SQL ORC Vectorized                                 1345           1357          17          7.8         128.3       6.2X
+SQL ORC MR                                         3012           3046          47          3.5         287.3       2.8X
+
+OpenJDK 64-Bit Server VM 11.0.14+9-LTS on Linux 5.11.0-1028-azure
+Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
 String with Nulls Scan (95.0%):           Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-SQL CSV                                            4443           4504          86          2.4         423.7       1.0X
-SQL Json                                           4528           4563          49          2.3         431.8       1.0X
-SQL Parquet Vectorized: DataPageV1                  213            233          15         49.2          20.3      20.8X
-SQL Parquet Vectorized: DataPageV2                  267            294          22         39.3          25.4      16.7X
-SQL Parquet MR: DataPageV1                         1691           1700          13          6.2         161.2       2.6X
-SQL Parquet MR: DataPageV2                         1515           1565          70          6.9         144.5       2.9X
-ParquetReader Vectorized: DataPageV1                228            231           2         46.0          21.7      19.5X
-ParquetReader Vectorized: DataPageV2                285            296           9         36.8          27.1      15.6X
-SQL ORC Vectorized                                  369            425          82         28.4          35.2      12.1X
-SQL ORC MR                                         1457           1463           9          7.2         138.9       3.0X
+SQL CSV                                            6169           6176          10          1.7         588.3       1.0X
+SQL Json                                           5352           5376          35          2.0         510.4       1.2X
+SQL Parquet Vectorized: DataPageV1                  248            255           7         42.3          23.6      24.9X
+SQL Parquet Vectorized: DataPageV2                  364            372          10         28.8          34.7      17.0X
+SQL Parquet MR: DataPageV1                         1624           1626           3          6.5         154.9       3.8X
+SQL Parquet MR: DataPageV2                         1520           1526           8          6.9         145.0       4.1X
+ParquetReader Vectorized: DataPageV1                259            262           1         40.4          24.7      23.8X
+ParquetReader Vectorized: DataPageV2                376            378           2         27.9          35.9      16.4X
```

Comment:
Hmm this still shows a degradation (69% vs 80% originally). I wonder why vectorized reader doesn't help here.

Reply:
Which line(s) are you looking at? Not sure which numbers are 69%, and 80% that you're referring to.

One reason vectorization isn't helping (maybe) is because the calling code is calling the vectorized reader one row at at time. 
(https://github.com/apache/spark/blob/439975590cf4f21c2a548a2ac6231eb234e1a2f3/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java#L239)
Let me address this in another PR. (https://issues.apache.org/jira/browse/SPARK-38254)

Reply:
I was looking at numbers of "ParquetReader Vectorized: DataPageV2" vs "ParquetReader Vectorized: DataPageV1". Previously it is 15.6/19.5 = 0.8, while the new ratio is 16.4/23.8 = 0.69. The other two cases (0% and 50% nulls) look OK though.